lib/cloud_pub_sub/adapters/tortoise/ssl.ex

```diff
@@ -54,13 +23,53 @@ defmodule CloudPubSub.Adapters.Tortoise.SSL do
       server: server
     }
 
-    connection_opts =
-      case opts[:cloud_provider] do
-        :aws -> connection_opts
-        :gcp -> Map.put(connection_opts, :password, opts[:password])
-      end
+    case opts[:cloud_provider] do
+      :aws ->
```

Comment:
(optional)

Something to think about: Rather than switching on cloud provider throughout the code, what if the sections that needed provider-specific implementation called out to a provider's module (i.e. `CloudPubSub.Provider.AWS`)? This may help to locate all the config for a provider in one place, and make it easier to add new providers in the future.

Reply:
I like it, could you make an issue asking for as much?

Reply:
https://github.com/verypossible-labs/cloud_pub_sub/issues/17